go/libraries/doltcore/sqle/dtables/column_diff_table.go

```diff
@@ -0,0 +1,656 @@
+	if err != nil {
+		return nil, err
+	}
+
+	return &tableColChange{
+		tableName: delta.ToName,
+		colNames:  colNames,
+		diffTypes: diffTypes,
+	}, nil
+}
+
+// calculateColDelta iterates through the rows of the given table delta and compares each cell in the to_ and from_
+// cells to compile a list of modified columns
+func calculateColDelta(ctx *sql.Context, ddb *doltdb.DoltDB, delta *diff.TableDelta, colSchemaDiff *ColSchemaDiff) ([]string, []string, error) {
+	// initialize row iterator
+	diffTableSchema, j, err := GetDiffTableSchemaAndJoiner(delta.ToTable.Format(), delta.FromSch, delta.ToSch)
+	if err != nil {
+		return nil, nil, err
+	}
+	diffTableCols := diffTableSchema.GetAllCols()
+
+	now := time.Now() // accurate commit time returned elsewhere
+	dp := NewDiffPartition(delta.ToTable, delta.FromTable, delta.ToName, delta.FromName, (*dtypes.Timestamp)(&now), (*dtypes.Timestamp)(&now), delta.ToSch, delta.FromSch)
+	ri := NewDiffPartitionRowIter(*dp, ddb, j)
+
+	var resultColNames []string
+	var resultDiffTypes []string
+	// add all added/dropped columns to result
+	for _, col := range colSchemaDiff.addedCols {
+		resultColNames = append(resultColNames, col)
+		resultDiffTypes = append(resultDiffTypes, diffTypeAdded)
+	}
+	for _, col := range colSchemaDiff.droppedCols {
+		resultColNames = append(resultColNames, col)
+		resultDiffTypes = append(resultDiffTypes, diffTypeRemoved)
+	}
+
+	colNamesSet := make(map[string]struct{})
+	// check each row for diffs in modified columns
+	for {
+		r, err := ri.Next(ctx)
+		if err == io.EOF {
+			for col := range colNamesSet {
+				// append modified columns to result
+				resultColNames = append(resultColNames, col)
+				resultDiffTypes = append(resultDiffTypes, diffTypeModified)
+			}
+			return resultColNames, resultDiffTypes, nil
+		} else if err != nil {
+			return nil, nil, err
+		}
+
+		// only need to check modified columns
+		for _, col := range colSchemaDiff.modifiedCols {
+			toColTag := diffTableCols.NameToCol["to_"+col].Tag
+			fromColTag := diffTableCols.NameToCol["from_"+col].Tag
+			toIdx := diffTableCols.TagToIdx[toColTag]
+			fromIdx := diffTableCols.TagToIdx[fromColTag]
+
+			if r[toIdx] != r[fromIdx] {
+				colNamesSet[col] = struct{}{}
+			}
+		}
+
+		// can stop checking rows when we already have all modified columns in the result set
+		if len(colNamesSet) == len(colSchemaDiff.modifiedCols) {
+			for col := range colNamesSet {
+				// append modified columns to result
+				resultColNames = append(resultColNames, col)
+				resultDiffTypes = append(resultDiffTypes, diffTypeModified)
+			}
+			return resultColNames, resultDiffTypes, nil
+		}
+	}
+}
+
+// ColSchemaDiff is a collection of column names that hold the results of doing a schema diff between to/from schemas,
+// i.e. a list of column names for each type of change, the total list of column names, and a corresponding list of
+// diff_types for each column
+type ColSchemaDiff struct {
```

Comment:
(minor) Try to avoid exporting (i.e. capitalizing) types unless you have to. It's nice to keep them package protected by default. In this case, `ColSchemaDiff` looks like it's only used in this file, so you could rename it to `colSchemaDiff` to keep it visible only in the current package.